### ISSUE TITLE
Type error for audit in version 4.4.0

### ISSUE THREAD

**Issue by ruegerj** [vlvtin/vlquery#18]
I attempted to lift the version of vlquery from `3.9.8` to the latest `4.4.0` for a project.

The following `vlconfig.json` is currently in use. The _virtual delete_ is enabled, alongside the _audittrail_.

```json
{
    "context": {
        "outFile": "src/db/db.context.ts",
        "runContext": true,
        "active": "_active",
        "connection": {
            "user": "<user>",
            "password": "<password>",
            "database": "<database>"
        },
        "audit": {
            "entity": "audit",
            "track": {
                "timestamp": "timestamp",
                "comment": "comment",
                "action": "action",
                "object": "object",
                "entity": "entity",
                "objectId": "id",
                "username": ["currentUser", "username"]
            },
            "commentRequired": true
        }
    }
}
```

The local db-context could be successfully re-generated. But when I try to build my project using `tsc` I face the following type error, which originates from the generated db-context:

![image](https://user-images.githubusercontent.com/47751318/124611761-09fac700-de72-11eb-828e-7e485c783806.png)

The part of the context, were the error originates from, is the following:

```typescript
DbSet.$audit = {
	table: "audit",
	contextRequired: true,
	commentRequired: true,

	async createAudit(action: "create" | "update" | "delete", comment: string, entity: Entity<any>, runContext?: any) {
		const audit = new Audit();
		audit.timestamp = new Date();
		audit.comment = comment;
		audit.action = action;
		const object = {};

		for (let key in entity.$$meta.columns) {
			object[key] = entity[key];
		}

		audit.object = object;
		audit.entity = entity.$$meta.tableName;
		audit.objectId = entity.id;
		audit.username = await (await runContext["currentUser"])["username"];

		return audit;
	}
}
```
I assume the issue is related to the composite type `string | number` defined by the `id` property of the  [Entity](https://github.com/levvij/vlquery/blob/master/src/entity.ts#L23) base class. Maybe some auto magic during context generation could fix the issue. One could keep track of the `objectId` type of the _audit_ entity during context generation and wrap the statement accordingly with a parsing function:

```typescript
// when 'objectId' is of type `string`
audit.objectId = entity.id.toString();

// when 'objectId' is of type 'number'
audit.objectId = parseInt(entity.id, 10);
```

By this the type safety would be restored and the type error would be resolved. Alternatively the type could be asserted during runtime, although I'm not to sure what a performance impact this could have. Therefore a static solution would be favorable in my opinion, because the context is generated anyways and the required information is at hand.

**Comment by levvij** [comment]
Will be fixed instantly!

We added number-Id support and forgot about this, sorry!

**Comment by ruegerj** [comment]
No worries, it isn't a dealbreaker for me. I can rely on my local version, which works perfectly fine :)
Thank you very much for your great support 👍

**Comment by levvij** [comment]
@ruegerj should be patched in 4.4.1, please close the issue if it works. 

Solved by converting the id to a string, because you can use both simultaneously in a context.

**Comment by ruegerj** [comment]
Works like a charm in `4.4.1`, thank you
